src/test/java/com/test/springbdd/ProductStepDefination.java

```diff
@@ -17,8 +19,9 @@
 import io.cucumber.java.en.When;
 
 @RunWith(SpringRunner.class)
-@WebMvcTest(ProductController.class)
-
+@SpringBootTest
```

Comment:
Change this to @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)